### PR TITLE
OCPBUGS#2012: Added Azure cloud provider clause

### PR DESCRIPTION
**Pull request by snarayan-redhat**
Version(s): 4.8+

Issue: https://issues.redhat.com/browse/OCPBUGS-2012

Link to docs preview: https://52809--docspreview.netlify.app/openshift-enterprise/latest/installing/installing-preparing.html#installing-preparing-install-manage

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by duanwei33** [comment]
@jsafrane Do you think we really need such info for dynamic provisioning, or we could put in a common place to mention it?

**Comment by kcarmichael08** [comment]
Hi Shubha, this LGTM, noting that QE has a question above, I think?

**Comment by jsafrane** [comment]
It's not only about dynamic provisioning, we support the whole AzureDisk / AzureFile storage only on Azure cloud or Azure Stack Hub. We do not test on bare metal clusters with SMB / CIFS shares.

**Comment by jsafrane** [comment]
(BTW, this applies to all cloud-based storage backends)

**Comment by snarayan-redhat** [comment]
@duanwei33 - If this is not specific to dynamic provisioning as @jsafrane mentioned, it should probably stay in a common place. How about mentioning it in  the introduction?

**Comment by duanwei33** [comment]
In any platforms like AWS/Azure/GCE/vSphere/OSP/others, if user creates vm and install cluster as baremetal, it has the same limitation. (As @jsafrane mentioned) 
So I don't think adding in storage/persistent_storage/persistent-storage-azure section is a good place. But currently I don't have a good place. Maybe in "Storage overview" or "baremetal installation"?

**Comment by snarayan-redhat** [comment]
I would prefer Baremetal installation. I think users may or may not read Storage overview. The probability of users going through a BM installation doc is higher. Also I see this potential place to mention it. Does it look like a logical place to present this info? https://docs.openshift.com/container-platform/4.11/installing/installing-preparing.html#installing-preparing-selecting-cluster-type

**Comment by duanwei33** [comment]
@snarayan-redhat Yes, I think that is a better place than Storage overview, it should be the very beginning to start Openshift. I'm okay with that. Not sure if others have any comments.

**Comment by snarayan-redhat** [comment]
@duanwei33 - So in the above location I will mention that cloud-based storage is not supported with BM installation. Does this sound good?

**Comment by snarayan-redhat** [comment]
@duanwei33 - need your final approval on this change.

File: installing/installing-preparing.adoc [hunk 1]
```diff
@@ -39,7 +39,7 @@ endif::openshift-origin[]
 
 You can deploy an {product-title} 4 cluster to both on-premise hardware and to cloud hosting services, but all of the machines in a cluster must be in the same data center or cloud hosting service.
```
Comment (duanwei33) [inline installing/installing-preparing.adoc:40]: I think this is a better place for the new adding. 
{product-title} does not support cloud-based storage for bare metal installation.

@jsafrane Do you think this sentence make sense? Or any suggestion here?

Comment (jsafrane) [reply installing/installing-preparing.adoc:40]: Yes, this place sounds a bit better to me, but it's not much of a difference.

File: installing/installing-preparing.adoc [hunk 2]
```diff
@@ -39,7 +39,7 @@ endif::openshift-origin[]
...
 
-If you want to use {product-title} but do not want to manage the cluster yourself, you have several managed service options. If you want a cluster that is fully managed by Red Hat, you can use link:https://www.openshift.com/products/dedicated/[OpenShift Dedicated] or link:https://www.openshift.com/products/online/[OpenShift Online]. You can also use OpenShift as a managed service on Azure, AWS, IBM Cloud VPC, or Google Cloud. For more information about managed services, see the link:https://www.openshift.com/products[OpenShift Products] page.
+If you want to use {product-title} but do not want to manage the cluster yourself, you have several managed service options. If you want a cluster that is fully managed by Red Hat, you can use link:https://www.openshift.com/products/dedicated/[OpenShift Dedicated] or link:https://www.openshift.com/products/online/[OpenShift Online]. You can also use OpenShift as a managed service on Azure, AWS, IBM Cloud VPC, or Google Cloud. For more information about managed services, see the link:https://www.openshift.com/products[OpenShift Products] page. {product-title} does not support cloud-based storage for bare metal installation.
```
Comment (jsafrane) [inline installing/installing-preparing.adoc:42]: I'm not a native speaker and the new sentence does not make much sense to me. Why would I use cloud storage on a real on prem bare metal OCP installation? Can we somehow emphasize that this targets clusters that use a cloud VM as a (virtual) bare metal?

Comment (snarayan-redhat) [reply installing/installing-preparing.adoc]: @jsafrane Can you verify if the reworded statement sounds better?

Comment (jsafrane) [reply installing/installing-preparing.adoc]: lgtm

File: installing/installing-preparing.adoc [hunk 3]
```diff
@@ -40,7 +40,7 @@ endif::openshift-origin[]
...
+If you want to use {product-title} but do not want to manage the cluster yourself, you have several managed service options. If you want a cluster that is fully managed by Red Hat, you can use link:https://www.openshift.com/products/dedicated/[OpenShift Dedicated] or link:https://www.openshift.com/products/online/[OpenShift Online]. You can also use OpenShift as a managed service on Azure, AWS, IBM Cloud VPC, or Google Cloud. For more information about managed services, see the link:https://www.openshift.com/products[OpenShift Products] page. If you install an {product-title} cluster with a cloud virtual machine as a virtual bare metal, the corresponding cloud-based storage is not supported.
```
Comment (mburke5678) [inline installing/installing-preparing.adoc:43]: @snarayan-redhat I am not sure what _as a virtual bare metal_ means.

Also, I'm not an exper here, but I wonder if we should tell people, as the Jira suggests, what they need to do to if they want cloud-based storage. 

> the Dynamic Provisioning will only work if we did Azure cloud provider (IPI/UPI) installation and not the baremetal/UPI installation.

Comment (snarayan-redhat) [reply installing/installing-preparing.adoc:43]: @mburke5678 I was on the same page as you. This was suggested and approved by SME and also double confirmed with QE on the term's validity.  Also regarding the ticket, the context of the issue went beyond dynamic provisioning. The bug fix started with the Jira's context, and as per SME's comment, it applied to OCP in whole.

**Pull request merged**